app/javascript/components/widget-controller.jsx

```diff
@@ -71,19 +79,38 @@ export class WidgetController extends React.Component {
 
   moveDown = () => {
     this.setState(({ index }) => ({
-      index: (index + 1) % widgets.length,
+      index: (index + 1) % this.visibleWidgets.length,
     }));
   };
 
   moveUp = () => {
     this.setState(({ index }) => ({
-      index: mathMod(index - 1, widgets.length),
+      index: mathMod(index - 1, this.visibleWidgets.length),
     }));
   };
 
+  setVisibleWidgets = () => {
+    const date = new Date();
+    const currentHour = date.getHours();
+
+    return widgets.filter(widget => {
+      if (widget.timeStart) {
+        if (widget.timeStart <= currentHour) {
+          if (widget.timeEnd >= currentHour) {
+            return true;
+          }
+        }
+        return false;
+      }
+      return true;
+    });
+  };
+
   render() {
     const { index } = this.state;
-    const currentWidget = widgets[index];
+    this.visibleWidgets = this.setVisibleWidgets();
```

Comment:
if we call `setVisibleWidgets()` on render we don't really need to do it anywhere else.

Probably we could rename it to `getVisibleWidgets` and skip storing it in the class instance also.